### ISSUE TITLE
Incoherent behavior between generator and controller while searching for endpoints

### ISSUE THREAD

**Issue by cromoteca** [vaadin/hilla#697]
### Description of the bug 

Endpoints are searched in all packages and translated to TypeScript, but then, when the application is executed, The endpoint registry in the controller is populated using `getBeansWithAnnotation` which only searches the Spring Boot application package by default. So, any endpoint outside of that is available when writing code, but not at execution time.

### Minimal reproducible example

Create a Hilla application and move `HelloWorldEndpoint` outside of `com.example.application`.

### Expected behavior

Either the endpoint is still accessible, or it is not translated to TypeScript at all.

### Actual behavior

TS is generated, but a 404 is returned at any invocation.

### Versions
Basically any recent version. Changing generator makes no difference.

### Workaround
Tell Spring to search inside the new package:

```java
@SpringBootApplication(scanBasePackages = { "com.example.application", "com.example.myendpoints" })
```

**Comment by cromoteca** [comment]
In the controller, consider using `ClassPathScanningCandidateComponentProvider` which can also be limited to a single package if needed:

```java
var scanner = new ClassPathScanningCandidateComponentProvider(false);
scanner.addIncludeFilter(new AnnotationTypeFilter(Endpoint.class));
var result = scanner.findCandidateComponents("mypackage");
```

**Comment by platosha** [comment]
Would be great if the runtime and the generator package filters are defined from a single source of truth.

**Comment by platosha** [comment]
Another thing, it would make sense to require this whitelist definition by default, to avoid any ambiguity regarding detection of the main package.

**Comment by platosha** [comment]
Using configuration extracted from Spring annotations seems complicated to implement in compile time. In addition, Spring Boot configuration serves another purpose. Let’s use a separate whitelist shared between the parser and the runtime.

Now need to define where to put the whitelist. Any ideas / prototypes are very welcome.

**Comment by platosha** [comment]
Note that the default endpoint search scope should be aligned with the Spring application scope.
